packages/cloud_functions/test/cloud_functions_test.dart

```diff
@@ -32,25 +32,37 @@ void main() {
           .call(functionName: 'qux', parameters: <String, dynamic>{
         'quux': 'quuz',
       });
+      await CloudFunctions.instance
+          .call(functionName: 'buz', region: 'us-east1');
       expect(
         log,
         <Matcher>[
           isMethodCall(
             'CloudFunctions#call',
             arguments: <String, dynamic>{
               'functionName': 'baz',
+              'region': 'us-central1',
               'parameters': null,
             },
           ),
           isMethodCall(
             'CloudFunctions#call',
             arguments: <String, dynamic>{
               'functionName': 'qux',
+              'region': 'us-central1',
               'parameters': <String, dynamic>{
                 'quux': 'quuz',
               },
             },
           ),
+          isMethodCall(
+            'CloudFunctions#call',
+            arguments: <String, dynamic>{
+              'functionName': 'buz',
+              'region': 'us-east1',
+              'parameters': null,
+            },
+          ),
```

Comment:
Would be good to add some expected failure cases. Like not passing a region, or passing an invalid region.

Reply:
A test case for not passing a region is already there. 

> @Knupper I checked `FirebaseApp` hoping we got a default region for the project through in `google-services.json` but we don't. With Firestore now going multi-region it might be worth combining efforts here if there is a more direct route.
> 
> Other change that might be worth making is that FirebaseApp should be able to be passed in. This might already be in another PR. Might be worth making another PR for that however.
> 
> References to how Firestore plugin does it.
> https://github.com/flutter/plugins/blob/master/packages/cloud_firestore/lib/src/firestore.dart#L132
> https://github.com/flutter/plugins/blob/master/packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java#L80
> https://firebase.google.com/docs/reference/android/com/google/firebase/functions/FirebaseFunctions.html#getInstance%28com.google.firebase.FirebaseApp%2C+java.lang.String%29

Not sure if this should be part of this merge request. All FirebaseApp calls are not implemented yet. So i would move it into a seperate merge request to keep this as simple as possible.